src/components/geometry.js

```diff
@@ -61,12 +61,10 @@ module.exports.Component = registerComponent('geometry', {
 
   /**
    * Update geometry component schema based on geometry type.
-   *
-   * @param {object} data - New data passed by Component.
    */
   updateSchema: function (data) {
+    var currentGeometryType = this.oldData && this.oldData.primitive;
```

Comment:
unrelated to PR?

Reply:
yes, it checks for oldData rather than data